### ISSUE TITLE
CIEL v2024-10-04 release

### ISSUE THREAD

**Issue by bmamlin** [OpenConceptLab/ocl_issues#1938]
- [x] Validate CIEL on cieladmin
- [x] Generate zip file
- [x] Upload zip file to [liquibaserunner](https://github.com/openmrs/openmrs-contrib-liquibaserunner)
- [x] Create any new sources
- [x] Convert to JSON

### Staging
- [x] Create diff
- [x] Retire any deleted mappings
- [x] Bulk import
- [x] Validate
- [x] Release

### Production
- [x] Create diff
- [x] Retire any deleted mappings
- [x] Bulk import
- [x] Validate
- [x] Release

**Comment by bmamlin** [comment]
CIEL v2024-10-04 validated and released on Staging. No failures on import.

Production import appeared to stall at 285956/290272 (98.5%). @snyaggarwal reported it had completed. Import reported 40 failures (the same we've had where some Spanish non-preferred synonyms match other concept FSNs, since #1921 has not yet been deployed to production). On validation, there was one discrepancy: [Gloves, latex (169370)](https://app.openconceptlab.org/#/orgs/CIEL/sources/CIEL/concepts/169370/) was missing. Looking at the report, this concept was in the list of failures with the error message:  "Concept fully specified name must be unique for same source and locale: Guantes de goma de látex (locale: es, preferred: yes)".

I tried manually importing this one concept, expecting it to fail again and I'd need to manually delete the conflicting synonym to get it in... but I was surprised that the single concept import succeeded. Why would it fail during bulk import and then succeed when imported by itself? Maybe concept validation isn't performed during a manual import of text (instead of a file)? That doesn't seem likely. When I look for other concepts with "Guantes" in their names, I can't find one. So, maybe it was a bogus validation failure during the initial bulk import? Here's the entry from the report:

```json
      {
        "retired": false,
        "datatype": "N/A",
        "type": "Concept",
        "concept_class": "Misc",
        "source": "CIEL",
        "extras": {},
        "names": [
          {
            "locale": "en",
            "external_id": "bd51462c-d51d-4959-a085-b8651e93bfd0",
            "locale_preferred": true,
            "name": "Gloves, latex",
            "name_type": "FULLY_SPECIFIED"
          },
          {
            "locale": "es",
            "external_id": "87f3dabc-1473-41e6-8e78-4f5be5b09a65",
            "locale_preferred": false,
            "name": "Guantes de látex de goma",
            "name_type": ""
          },
          {
            "locale": "es",
            "external_id": "60831581-d96d-4385-8c82-701721d6ffc2",
            "locale_preferred": true,
            "name": "Guantes de goma de látex",
            "name_type": "FULLY_SPECIFIED"
          }
        ],
        "owner": "CIEL",
        "owner_type": "Organization",
        "external_id": "1d8ff472-7758-410f-aef6-51d8aaecff1c",
        "id": "169370",
        "descriptions": [],
        "errors": {
          "names": [
            "Concept fully specified name must be unique for same source and locale: Guantes de goma de látex (locale: es, preferred: yes)"
          ]
        }
      }
```

Strange. It doesn't make sense to me. In any case, now that 169370 exists on production, I've triggered a new export to validate again. Hopefully #1921 will make it to production, soon. 🙂

**Comment by bmamlin** [comment]
CIEL v2024-10-04 validated and released on staging and production.

**Comment by bmamlin** [comment]
@snyaggarwal deployed #1921 to production, so I scanned the 40 resources that failed to update for differences in names between staging & production (n=17). These and now updated in HEAD on Production.

<details><summary>Expand to see import used for 17 concepts.</summary>

```json
{"retired":false,"datatype":"Numeric","type":"Concept","concept_class":"Test","source":"CIEL","extras":{"units":"copies/ml","low_absolute":0.0,"allow_decimal":false},"names":[{"locale":"en","external_id":"898BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"HIV viral load","name_type":"FULLY_SPECIFIED"},{"locale":"fr","external_id":"899BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"VIH charge virale","name_type":"FULLY_SPECIFIED"},{"locale":"sw","external_id":"900BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Kipimo cha kujua  wingi wa virusi  vya ukimwi mwilini","name_type":"FULLY_SPECIFIED"},{"locale":"en","external_id":"100036BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":false,"name":"Viral load","name_type":""},{"locale":"fr","external_id":"144646BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":false,"name":"Charge virale VIH","name_type":""},{"locale":"es","external_id":"51deb60e-5106-4ec9-b78f-9d3886e1a701","locale_preferred":false,"name":"Prueba de carga viral de HIV","name_type":""},{"locale":"es","external_id":"e590817a-e31c-44cf-a860-6f20d0417e7f","locale_preferred":true,"name":"Prueba de carga viral de virus de la inmunodeficiencia humana","name_type":"FULLY_SPECIFIED"},{"locale":"en","external_id":"247a331e-289f-49a8-83ac-1cd793dd3a0e","locale_preferred":false,"name":"HIV viral load, quantitative","name_type":""}],"owner":"CIEL","owner_type":"Organization","external_id":"856AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA","id":"856","descriptions":[{"locale":"en","external_id":"858FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF","description":"This is a measure of the number of copies/ml of DNA/RNA in patients with HIV."},{"locale":"fr","external_id":"859FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF","description":"CHARGE VIRALE VIH"}]}
{"retired":false,"datatype":"N/A","type":"Concept","concept_class":"Symptom/Finding","source":"CIEL","extras":{},"names":[{"locale":"en","external_id":"2331BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Weakness","name_type":"FULLY_SPECIFIED"},{"locale":"fr","external_id":"137326BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Faiblesse","name_type":"FULLY_SPECIFIED"},{"locale":"ht","external_id":"137327BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"feblès","name_type":"FULLY_SPECIFIED"},{"locale":"en","external_id":"143188BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":false,"name":"Weakness, general","name_type":""},{"locale":"es","external_id":"bee62e99-04e8-4b93-b368-c9d91d2a37b5","locale_preferred":false,"name":"Astenia","name_type":""},{"locale":"es","external_id":"0bb98cc8-39ce-4762-b2e7-352f798bdb2f","locale_preferred":false,"name":"Debilidad general","name_type":""},{"locale":"es","external_id":"73160619-6eee-4d04-b880-5ca8d3fef606","locale_preferred":false,"name":"Desgano","name_type":""},{"locale":"es","external_id":"2142eaa3-91bd-4402-899d-e12db492d010","locale_preferred":false,"name":"Abatimiento","name_type":""},{"locale":"es","external_id":"1af3fc25-d59f-475a-b251-1778fb02846f","locale_preferred":true,"name":"Debilidad","name_type":"FULLY_SPECIFIED"}],"owner":"CIEL","owner_type":"Organization","external_id":"5226AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA","id":"5226","descriptions":[{"locale":"en","external_id":"1850FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF","description":"The condition or quality of being weak."}]}
{"retired":false,"datatype":"N/A","type":"Concept","concept_class":"Diagnosis","source":"CIEL","extras":{},"names":[{"locale":"en","external_id":"14542BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Post-dysenteric reactive arthropathy","name_type":"FULLY_SPECIFIED"},{"locale":"es","external_id":"85084BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Artropatía posdisentérica","name_type":"FULLY_SPECIFIED"}],"owner":"CIEL","owner_type":"Organization","external_id":"113942AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA","id":"113942","descriptions":[]}
{"retired":false,"datatype":"N/A","type":"Concept","concept_class":"Diagnosis","source":"CIEL","extras":{},"names":[{"locale":"en","external_id":"14577BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Polymyalgia rheumatica","name_type":"FULLY_SPECIFIED"},{"locale":"es","external_id":"85067BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Polimialgia reumática","name_type":"FULLY_SPECIFIED"},{"locale":"ru","external_id":"132966BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Ревматическая полимиалгия","name_type":"FULLY_SPECIFIED"},{"locale":"es","external_id":"d74c83ee-ab70-4e49-8432-88d286d0fd3a","locale_preferred":false,"name":"Artritis senil","name_type":""},{"locale":"en","external_id":"aa6df4f0-f910-4801-9214-6e2701a944f9","locale_preferred":false,"name":"Senile arthritis","name_type":""}],"owner":"CIEL","owner_type":"Organization","external_id":"113977AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA","id":"113977","descriptions":[{"locale":"en","external_id":"3889FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF","description":"A syndrome in the elderly characterized by proximal joint and muscle pain, high erythrocyte sedimentation rate, and a self-limiting course. Pain is usually accompanied by evidence of an inflammatory reaction. Women are affected twice as commonly as men and Caucasians more frequently than other groups. The condition is frequently associated with TEMPORAL ARTERITIS and some theories pose the possibility that the two diseases arise from a single etiology or even that they are the same entity."}]}
{"retired":false,"datatype":"N/A","type":"Concept","concept_class":"Diagnosis","source":"CIEL","extras":{},"names":[{"locale":"en","external_id":"18410BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Bone fracture","name_type":"FULLY_SPECIFIED"},{"locale":"es","external_id":"82861BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Fractura ósea","name_type":"FULLY_SPECIFIED"}],"owner":"CIEL","owner_type":"Organization","external_id":"118055AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA","id":"118055","descriptions":[{"locale":"en","external_id":"5282FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF","description":"Fractures in which the break in bone is not accompanied by an external wound."}]}
{"retired":false,"datatype":"N/A","type":"Concept","concept_class":"Diagnosis","source":"CIEL","extras":{},"names":[{"locale":"en","external_id":"33540BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Muscle pain","name_type":"FULLY_SPECIFIED"},{"locale":"en","external_id":"126426BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":false,"name":"Muscle ache","name_type":""},{"locale":"fr","external_id":"139048BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Douleur musculaire","name_type":"FULLY_SPECIFIED"},{"locale":"ht","external_id":"139049BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"doulè nan misk","name_type":"FULLY_SPECIFIED"},{"locale":"es","external_id":"ca1867eb-8102-4119-b8c4-066ab16f2ba8","locale_preferred":true,"name":"Dolor muscular","name_type":"FULLY_SPECIFIED"}],"owner":"CIEL","owner_type":"Organization","external_id":"133632AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA","id":"133632","descriptions":[]}
{"retired":false,"datatype":"N/A","type":"Concept","concept_class":"Diagnosis","source":"CIEL","extras":{},"names":[{"locale":"en","external_id":"44328BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Clostridium difficile Infection","name_type":"FULLY_SPECIFIED"},{"locale":"es","external_id":"61261BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Infección por Clostridium difficile","name_type":"FULLY_SPECIFIED"},{"locale":"vi","external_id":"128349BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Nhiễm Clostridium difficile","name_type":"FULLY_SPECIFIED"},{"locale":"en","external_id":"da55e3bd-3ae0-4a2f-8730-e0577f0e9f36","locale_preferred":false,"name":"C. difficile Infection","name_type":""}],"owner":"CIEL","owner_type":"Organization","external_id":"144685AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA","id":"144685","descriptions":[]}
{"retired":false,"datatype":"N/A","type":"Concept","concept_class":"Drug","source":"CIEL","extras":{},"names":[{"locale":"en","external_id":"53927BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Adrenalin","name_type":"FULLY_SPECIFIED"},{"locale":"es","external_id":"0455437b-eed6-412c-9697-4b02e4f9e0fa","locale_preferred":false,"name":"Preparado de adrenalina","name_type":""},{"locale":"es","external_id":"ca899dc4-04df-46eb-a226-3d000871885f","locale_preferred":false,"name":"Preparado de epinefrina","name_type":""},{"locale":"es","external_id":"93e11229-fde3-4825-98a4-45aa98e7b4bc","locale_preferred":false,"name":"Preparado con epinefrina","name_type":""},{"locale":"es","external_id":"b83f9afd-f7a3-454f-b14b-a8cd099401eb","locale_preferred":false,"name":"Preparado con adrenalina","name_type":""},{"locale":"es","external_id":"5e1caa65-3501-4a26-a216-51a59ca0c4ca","locale_preferred":false,"name":"Epinefrina","name_type":""},{"locale":"es","external_id":"8df2b872-a2b2-41e8-b4ef-68ae79459564","locale_preferred":true,"name":"Adrenalina","name_type":"FULLY_SPECIFIED"}],"owner":"CIEL","owner_type":"Organization","external_id":"154365AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA","id":"154365","descriptions":[]}
{"retired":false,"datatype":"N/A","type":"Concept","concept_class":"Diagnosis","source":"CIEL","extras":{},"names":[{"locale":"en","external_id":"102338BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Ectodermosis erosiva pluriorificialis","name_type":"FULLY_SPECIFIED"},{"locale":"es","external_id":"ab237980-bcc2-42f6-a7eb-75e5a6a146a1","locale_preferred":true,"name":"Ectodermosis erosiva pluriorificial","name_type":"FULLY_SPECIFIED"}],"owner":"CIEL","owner_type":"Organization","external_id":"156110AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA","id":"156110","descriptions":[]}
{"retired":false,"datatype":"N/A","type":"Concept","concept_class":"Diagnosis","source":"CIEL","extras":{},"names":[{"locale":"en","external_id":"102394BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Erythema multiforme bullosum","name_type":"FULLY_SPECIFIED"},{"locale":"es","external_id":"2e7d0bfe-a71e-4d14-9eda-8fec181fecaa","locale_preferred":true,"name":"Eritema multiforme bulloso","name_type":"FULLY_SPECIFIED"}],"owner":"CIEL","owner_type":"Organization","external_id":"156166AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA","id":"156166","descriptions":[]}
{"retired":false,"datatype":"N/A","type":"Concept","concept_class":"Diagnosis","source":"CIEL","extras":{},"names":[{"locale":"en","external_id":"102395BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Erythema multiforme exudativum","name_type":"FULLY_SPECIFIED"},{"locale":"es","external_id":"03edce7b-bfa8-4779-a462-acc3118e4420","locale_preferred":true,"name":"Eritema multiforme exudativo","name_type":"FULLY_SPECIFIED"}],"owner":"CIEL","owner_type":"Organization","external_id":"156167AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA","id":"156167","descriptions":[]}
{"retired":false,"datatype":"N/A","type":"Concept","concept_class":"Diagnosis","source":"CIEL","extras":{},"names":[{"locale":"en","external_id":"102396BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Erythema multiforme major","name_type":"FULLY_SPECIFIED"},{"locale":"nl","external_id":"117184BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"syndroom van Stevens-Johnson","name_type":"FULLY_SPECIFIED"},{"locale":"vi","external_id":"127744BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Hồng ban đa dạng có bóng nước","name_type":"FULLY_SPECIFIED"},{"locale":"es","external_id":"1dc65803-f8b1-4809-9dbb-660257cab757","locale_preferred":true,"name":"Eritema multiforme mayor","name_type":"FULLY_SPECIFIED"}],"owner":"CIEL","owner_type":"Organization","external_id":"156168AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA","id":"156168","descriptions":[]}
{"retired":false,"datatype":"N/A","type":"Concept","concept_class":"Diagnosis","source":"CIEL","extras":{},"names":[{"locale":"en","external_id":"102432BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Extrapulmonary tuberculosis","name_type":"FULLY_SPECIFIED"},{"locale":"es","external_id":"98f57885-1ce6-4146-ad5d-9ae357dd6dfd","locale_preferred":false,"name":"TB extrapulmonar","name_type":""},{"locale":"es","external_id":"ba31d1d8-d1b2-44ca-93f3-74b773ae6b9d","locale_preferred":true,"name":"Tuberculosis extrapulmonar","name_type":"FULLY_SPECIFIED"}],"owner":"CIEL","owner_type":"Organization","external_id":"156204AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA","id":"156204","descriptions":[]}
{"retired":false,"datatype":"N/A","type":"Concept","concept_class":"Diagnosis","source":"CIEL","extras":{},"names":[{"locale":"en","external_id":"103582BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"intraventricular (nontraumatic) hemorrhage, grade 3, of fetus and newborn","name_type":"FULLY_SPECIFIED"},{"locale":"es","external_id":"a6485ed2-9736-4844-b93b-f002134639cc","locale_preferred":false,"name":"Hemorragia intraventricular (no traumática), grado 3, feto y recién nacido","name_type":""},{"locale":"es","external_id":"aa23be6f-8922-4a4d-9ee4-5c686fd37b3a","locale_preferred":true,"name":"hemorragia intraventricular (no traumática), grado 3, del feto y recién nacido","name_type":"FULLY_SPECIFIED"}],"owner":"CIEL","owner_type":"Organization","external_id":"157354AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA","id":"157354","descriptions":[]}
{"retired":false,"datatype":"N/A","type":"Concept","concept_class":"Diagnosis","source":"CIEL","extras":{},"names":[{"locale":"en","external_id":"104760BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Primary malignant neoplasm of vagina","name_type":"FULLY_SPECIFIED"},{"locale":"nl","external_id":"122801BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"maligne neoplasma van de vagina","name_type":"FULLY_SPECIFIED"},{"locale":"fr","external_id":"123409BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Tumeur maligne du vagin","name_type":"FULLY_SPECIFIED"},{"locale":"es","external_id":"27b3866c-e747-43dd-a44a-5fde5c275aa8","locale_preferred":false,"name":"Neoplasia maligna primaria de la vagina","name_type":""}],"owner":"CIEL","owner_type":"Organization","external_id":"158532AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA","id":"158532","descriptions":[]}
{"retired":false,"datatype":"N/A","type":"Concept","concept_class":"Diagnosis","source":"CIEL","extras":{},"names":[{"locale":"en","external_id":"104843BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Queensland fever","name_type":"FULLY_SPECIFIED"},{"locale":"es","external_id":"0f837a9d-6d8a-47c7-9f75-51ed2c8d9275","locale_preferred":false,"name":"Leptospirosis","name_type":""},{"locale":"es","external_id":"670d3e0e-600c-4129-a204-f81e6d3709cc","locale_preferred":false,"name":"Infección por Leptospira","name_type":""},{"locale":"es","external_id":"6aaa89a8-f2e2-43bc-b290-0b1ad156d987","locale_preferred":false,"name":"Enfermedad de los trabajadores de los campos de arroz","name_type":""},{"locale":"es","external_id":"3d5cfe3a-7162-4b00-a0eb-1bf54022532a","locale_preferred":false,"name":"Infección debida a Leptospira","name_type":""},{"locale":"es","external_id":"43c8dbbf-5808-451f-84ba-b4900a3d8cb0","locale_preferred":true,"name":"Fiebre de Queensland","name_type":"FULLY_SPECIFIED"}],"owner":"CIEL","owner_type":"Organization","external_id":"158615AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA","id":"158615","descriptions":[]}
{"retired":false,"datatype":"N/A","type":"Concept","concept_class":"Misc","source":"CIEL","extras":{},"names":[{"locale":"en","external_id":"106470BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Manual laborer","name_type":"FULLY_SPECIFIED"},{"locale":"en","external_id":"106471BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":false,"name":"Manual laborer","name_type":"SHORT"},{"locale":"es","external_id":"108367BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB","locale_preferred":true,"name":"Obrero","name_type":"FULLY_SPECIFIED"}],"owner":"CIEL","owner_type":"Organization","external_id":"159390AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA","id":"159390","descriptions":[{"locale":"en","external_id":"16021FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF","description":"As a occupation of the patient."}]}
```
</details>

**Comment by bmamlin** [comment]
CIEL v2024-10-04 re-released on production after #1921 deployed and previously blocked concepts updated.
